src/H5I.c

```diff
@@ -556,8 +581,23 @@ H5I_clear_type(H5I_type_t type, hbool_t force, hbool_t app_ref)
     udata.force   = force;
     udata.app_ref = app_ref;
 
-    /* Attempt to free all ids in the type */
-    if (H5SL_try_free_safe(udata.type_ptr->ids, H5I__clear_type_cb, &udata) < 0)
+    if (H5I_IS_LIB_TYPE(type)) {
+        H5I_id_info_t *item = NULL;
+        H5I_id_info_t *tmp  = NULL;
+
+        /* This is a "delete-safe" iteration */
+        HASH_ITER(hh, udata.type_ptr->ids.hash_table, item, tmp)
+        {
```

Comment:
You need to add HASH_ITER to the .clang-format file so that it recognizes it's an iterator, add it to the ForEachMacros section

Reply:
Will do